tools/viewer/main.rs

```diff
@@ -39,13 +39,26 @@ struct Cli {
     /// Automatically watch the file system, and reload when it changes
     #[structopt(long)]
     auto_reload: bool,
+
+    /// Load properties from a json file ('-' for stdin)
+    #[structopt(long, name = "load data file")]
+    load_data: Option<std::path::PathBuf>,
+
+    /// Store properties values in a json file at exit ('-' for stdout)
+    #[structopt(long, name = "save data file")]
+    save_data: Option<std::path::PathBuf>,
 }
 
 thread_local! {static CURRENT_INSTANCE: std::cell::RefCell<Option<ComponentInstance>> = Default::default();}
 
 fn main() -> Result<()> {
     let args = Cli::from_args();
 
+    if args.auto_reload && (args.save_data.is_some()) {
+        eprintln!("Cannot pass both --auto-reload and --save-data");
```

Comment:
Why not? :)

Reply:
When do we save the properties?

Reply:
Before reloading, and then maybe load again with the new ui loaded?

Kind of a way to persist properties across ui edit cycles.

Reply:
(Btw, just a side comment - don’t let that hold you off from merging :)